ibis/backends/tests/test_generic.py

```diff
@@ -2281,6 +2393,34 @@ def test_substitute(backend):
     assert expr["subs_count"].execute()[0] == t.count().execute() // 10
 
 
+@pytest.mark.broken("clickhouse", reason="special case this and returns 'oops'")
+def test_value_cases_null(con):
+    """CASE x WHEN NULL never gets hit"""
+    e = ibis.literal(5).nullif(5).cases((None, "oops"), else_="expected")
+    assert con.execute(e) == "expected"
+
+
+@pytest.mark.broken("pyspark", reason="raises a ResourceWarning that we can't catch")
+def test_case(con):
+    # just to make sure that the deprecated .case() method still works
+    with pytest.warns(FutureWarning, match=".cases"):
+        assert con.execute(ibis.case().when(True, "yes").end()) == "yes"
```

Comment:
Do you actually need to call `execute` here? Doesn't the warning happening in the API call? If so, please remove any API calls that are not necessary for triggering the warning.

Reply:
you are right, thank you!

Reply:
oh but before I change this, can you comment at https://github.com/ibis-project/ibis/pull/9096/files/e0fa647ebed6e2fffbd549a68834f2dc8727330c#r1676148277? If we are going to just delete the builder API, then we don't need this entire test

Reply:
Actually, this test both 1. tests that FutureWarning happens but also 2. checks that the old API still returns the correct result. So if we keep this test at all (depending on our decision in https://github.com/ibis-project/ibis/pull/9096#discussion_r1676148277), then this .execute() should remain. Resolving this convo, but re-open if you disagree.